packages/wdio-allure-reporter/src/index.ts

```diff
@@ -211,7 +211,10 @@ class AllureReporter extends WDIOReporter {
     onTestPass() {
         attachConsoleLogs(this._consoleOutput, this._allure)
         if (this._options.useCucumberStepReporter) {
-            return this._allure.endStep('passed')
+            const suite = this._allure.getCurrentSuite()
+            if (suite && suite.currentStep instanceof Step) {
```

Comment:
Shouldn't `suite.currentStep` simply be undefined or null? This would simplify the check

Reply:
Thanks for taking a look! I left some details here - https://github.com/webdriverio/webdriverio/issues/6523#issuecomment-889823216

 If currentStep is of another type, say `Test` which I believe is the cause of this issue, (not sure why that happens though, maybe hooks? maybe only on async?), It sould still satisfy a null / undef check.

Thanks!